var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -215,9 +215,16 @@ def cmake_args(self):
 
         if '+cuda' in spec:
             args.append('-DQMC_CUDA=1')
-            cuda_arch = spec.variants['cuda_arch'].value
+            cuda_arch_list = spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
```

Comment:
This line seems to be duplicated below. What happens if no `cuda_arch` is specified? Does `cuda_arch_list[0]` crash?

Reply:
Not following your comment about duplication.

I found this super counter-intuitive, if no `cuda_arch` is specified, what we have is:
``cuda_arch_list = ('none',)``

No, there is no crash -- by construction the list is never `None`. I guest this make sense since the right-hand side is allowed to be multi-valued.

BTW, this documentation needs to get updated:
https://spack.readthedocs.io/en/latest/build_systems/cudapackage.html

Reply:
> Not following your comment about duplication.

If you do this:
```python
cuda_arch_list = spec.variants['cuda_arch'].value
cuda_arch = cuda_arch_list[0]
```
then you don't need to do this below:
```python
cuda_arch = spec.variants['cuda_arch'].value[0]
```
I think the minimal change would be to check the length of `cuda_arch`, and if it's equal to 1, use it, otherwise throw an error.

Reply:
oops, sorry I meant to delete that extra `cuda_arch` statement

I already throw an error if `cuda_arch_list` is greater than 1. Since there is no support for it in QMCPACK.